File: API/Model/ViewModel/ApprovalFlowViewMaster.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Reflection.Metadata;
using System.Text;
using System.Threading.Tasks;

namespace Model.ViewModel
{

    public class ApprovalFlowNewViewMaster
    {
        public string Id { get; set; } = string.Empty;
        public string SchemeId { get; set; } = string.Empty;
        public string RoleId { get; set; } = string.Empty;
        public int OrderNumber { get; set; } 
    }

    public class ApprovalFlowViewMaster
    {
        public string Id { get; set; } = string.Empty;
        public string RoleId { get; set; } = string.Empty;
        public int OrderNumber { get; set; }
        public bool IsNA { get; set; }
        public string ApprovalFlowId { get; set; } = string.Empty;
        public string ReturnFlowId { get; set; } = string.Empty;
        public bool IsFinal { get; set; }
        public bool IsActive { get; set; }
        public string RoleName { get; set; } = string.Empty;
        public string RoleCode { get; set; } = string.Empty;

        public string ApprovalFlowRoleName { get; set; } = string.Empty;
        public string ApprovalFlowRoleCode { get; set; } = string.Empty;

        public string ReturnFlowRoleName { get; set; } = string.Empty;
        public string ReturnFlowRoleCode { get; set; } = string.Empty;

        public string LastUpdatedBy { get; set; } = string.Empty;
        public string LastUpdatedUserName { get; set; } = string.Empty;
        public DateTime? LastUpdatedDate { get; set; }
    }

    public class ApprovalFlowAddRoleModel
    {
        public string SchemeId { get; set; } = string.Empty;
        public List<string> RoleIds { get; set; } = null!;
    }

    public class ApprovalFlowAddRoleModel_API
    {
        public string SchemeId { get; set; } = string.Empty;
        public List<string> RoleIds { get; set; } = null!;
        public string SavedBy { get; set; } = string.Empty;
        public string SavedByUserName { get; set; } = string.Empty;
        public DateTime SavedDate { get; set; }
    }
}

File: UI/src/app/_models/ApprovalFlowViewMaster.ts
export interface ApprovalFlowViewMaster {
  id: string;
  roleId: string;
  orderNumber: number;
  isNA: boolean;
  approvalFlowId: string;
  returnFlowId: string;
  isFinal: boolean;
  isActive: boolean;
  roleName: string;
  roleCode: string;
  approvalFlowRoleName: string;
  approvalFlowRoleCode: string;
  returnFlowRoleName: string;
  returnFlowRoleCode: string;
  lastUpdatedBy: string;
  lastUpdatedUserName: string;
  lastUpdatedDate: string | null;
}
export interface ApprovalFlowAddRoleModel {
  roleIds: string[];
  schemeId: string;
}
export interface ApprovalFlowNewViewMaster {
  id: string;
  roleId: string;
  schemeId: string;
  orderNumber: number;
}
export interface ApprovalFlowNewViewMaster {
  id: string;
  roleId: string;
  orderNumber: number;
}
export interface SchemeStatusMappingViewModel {
  schemeId: string;
  statusId: string;
  status: string;
  sortOrder: number;
  isDisabled: boolean;
}

export interface SchemeStatusMappingSaveModel {
  schemeId: string;
  statusIds: string[];
}

export interface ConfigSchemeStatusMappingModel {
  id: string;
  schemeId: string;
  statusId: string;
  sortOrder: number;
  schemeName: string;
  schemeCode: string;
  statusName: string;
  statusCode: string;
  isDisabled: boolean;
}
export interface DocumentGroupConfigurationOrderModel {
  schemeId: string;
  documentGroupId: string;
  documentGroupName: string;
  sortOrder: number;
}
export interface SchemeGroupOrderingModel {
  id: string;
  groupName: string;
  groupNameTamil: string;
  groupNameEnglish: string;
  sortOrder: number;
}

export interface SchemeOrderingModel {
  id: string;
  schemeName: string;
  schemeNameEnglish: string;
  schemeNameTamil: string;
  sortOrder: number;
}
export interface CardPrintStatusOrderModel {
  id: string;
  sortOrder: number;
}
